cloudinit/cmd/status.py

```diff
@@ -196,6 +196,54 @@ def get_bootstatus(disable_file, paths) -> Tuple[UXAppBootStatusCode, str]:
     return (bootstatus_code, reason)
 
 
+def _get_systemd_status() -> Optional[UXAppStatus]:
+    """Get status from systemd.
+
+    Using systemd, we can get more fine-grained status of the
+    individual unit. Determine if we're still
+    running or if there's an error we haven't otherwise detected
+    """
+    for service in [
+        "cloud-final.service",
+        "cloud-config.service",
+        "cloud-init.service",
+        "cloud-init-local.service",
+    ]:
+        stdout = subp.subp(
+            [
+                "systemctl",
+                "show",
+                "--property=ActiveState,UnitFileState,SubState",
+                service,
+            ],
+        ).stdout
+        states = dict(
+            [[x.strip() for x in r.split("=")] for r in stdout.splitlines()]
+        )
+        if not (
+            states["UnitFileState"].startswith("enabled")
+            or states["UnitFileState"] == "static"
+        ):
+            # Service is not set to run, so don't determine state
+            # based on this service
+            continue
```

Comment:
I think we should consider this an error condition because cloud-init boot stages need to be active and enabled, and run serially from local -> init -> config -> final for us to support cloud-init typical first boot config. The times we've have bugs filed that cloud-init didn't work (and nothing reported an error) was due to systemd ordering cycles resulting in cloud-init systemd units being kicked out of boot targets. I'd suggest if we are not in systemd generator "DISABLED" state, all systemd units need to be enabled for us to claim cloud-init is happy/supportable.